components/checkout/CheckoutSidebarView/CheckoutSidebarView.tsx

```diff
@@ -12,16 +12,24 @@ import PaymentWidget from '../PaymentWidget'
 import s from './CheckoutSidebarView.module.css'
 
 const CheckoutSidebarView: FC = () => {
+  const [loadingSubmit, setLoadingSubmit] = useState(false)
   const { setSidebarView, closeSidebar } = useUI()
-  const { data: cartData } = useCart()
+  const { data: cartData, revalidate: refreshCart } = useCart()
   const { data: checkoutData, submit: onCheckout } = useCheckout()
 
   async function handleSubmit(event: React.ChangeEvent<HTMLFormElement>) {
-    event.preventDefault()
+    try {
+      setLoadingSubmit(true)
+      event.preventDefault()
 
-    await onCheckout()
-
-    closeSidebar()
+      await onCheckout()
+      refreshCart()
+      closeSidebar()
+    } catch {
+      // TODO - handle error UI here.
+    } finally {
+      setLoadingSubmit(false)
```

Comment:
Wouldn't setting this to false after closing the sidebar cause a warning for setting a state after the component has been unmounted?

Reply:
Good spot - my console warning confirms that you're correct. Fixed now